plugins/ata/fu-ata-device.c

```diff
@@ -246,32 +381,62 @@ fu_ata_device_parse_id (FuAtaDevice *self, const guint8 *buf, gsize sz, GError *
 		fu_device_set_version_format (device, FWUPD_VERSION_FORMAT_PLAIN);
 	}
 
+	/* get OUI if set */
+	self->oui = ((guint32) (id[108] & 0x0fff)) << 12 |
+		    ((guint32) (id[109] & 0xfff0)) >> 4;
+	if (self->oui > 0x0) {
+		g_autofree gchar *tmp = NULL;
+		tmp = g_strdup_printf ("OUI\\%06x", self->oui);
+		fu_device_add_instance_id (device, tmp);
+		has_oui_quirk = fu_device_get_vendor (FU_DEVICE (self)) != NULL;
+	}
+
+	/* if not already set using the vendor block or a OUI quirk */
+	name = fu_ata_device_get_string (id, 27, 46);
+	if (name != NULL && !has_oui_quirk)
+		fu_ata_device_parse_vendor_name (self, name);
+
+	/* ask user to report data */
+	if (self->oui != 0x0 && !has_oui_quirk &&
```

Comment:
I would think you want to have this show up in `UpdateError` in some way.  And then you can point people to the wiki page, and the wiki page will point them to daemon journal logs with the details.

Reply:
Also, if you're going all in on these, you probably want to turn off the updatable flag in these instances too.

Reply:
> Also, if you're going all in on these, you probably want to turn off the updatable flag in these instances too.

If we do this the device "drops out" from `fwupdmgr get-devices` as it then needs `--show-all-devices` -- is this what we want? It means less people see the update message...

Reply:
> turn off the updatable flag

My StarLabs drive doesn't have a WWN, but is updatable as it have firmware on the LVFS. Or would we only remove UPDATABLE if the vendor_id is NULL, i.e. the hueristic failed also? I guess it depends if we want UPDATABLE to mean "the hardware is updatable" or "the policy allows this device to be updated"

Reply:
> If we do this the device "drops out" from fwupdmgr get-devices as it then needs --show-all-devices -- is this what we want? It means less people see the update message...

Not if you set `UpdateError`.  It's listed then still.

> My StarLabs drive doesn't have a WWN, but is updatable as it have firmware on the LVFS. Or would we only remove UPDATABLE if the vendor_id is NULL, i.e. the hueristic failed also? I guess it depends if we want UPDATABLE to mean "the hardware is updatable" or "the policy allows this device to be updated"

I think removing when vendor ID is NULL makes most sense.

Reply:
> Not if you set UpdateError.

Do you mind if I do that as a phase2? I'd like only technical users to see this "report" quirk for the first few weeks; when we have the ~12 most popular SSD vendors covered I can open it up a bit.

Reply:
> I think removing when vendor ID is NULL makes most sense.

Done.

Reply:
> Do you mind if I do that as a phase2? I'd like only technical users to see this "report" quirk for the first few weeks; when we have the ~12 most popular SSD vendors covered I can open it up a bit.

Realistically I don't think anyone is going to see this if it's only in the journal output.  Setting `UpdateError` will at least tell you to go look at the journal.

The other thing you can do is automatically generate the quirk and drop it in `/var/cache`, and the wiki page tell someone to go pick it up.

Reply:
> automatically generate the quirk

In many cases we don't know the vendor ID or name. The lack of vendor-id is now put in the update message, but no link to the wiki. I'm really hesitant to put one there after seeming people email the flashrom mailing list with years-old versions of flashrom data.

Reply:
OK.. I think the `UpdateError` was the most important part.  One more thing, rather than an environment variable can you move the toggle for this into a `ata.conf` plugin configuration file?

Reply:
I think that's a good idea, but would need something like https://github.com/fwupd/fwupd/pull/1790 first

Reply:
Okay, fixed I think.